### ISSUE TITLE
[Enhancement] Non-linear font scaling

### ISSUE THREAD

**Issue by dennispedersen-bd** [kirbydesign/designsystem#4200]
- [ ] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement

POC on scaling fonts non-linear, so it is possible to view more text while scaling. This is the approach taken by native apps on Android and iPhone.

See ongoing dialog with WCAG regarding this and more details here:
https://github.com/w3c/matf/issues/3 

Apple font scaling concept:
https://developer.apple.com/design/human-interface-guidelines/typography#Specifications

Apple Dev:
https://developer.apple.com/documentation/uikit/scaling-fonts-automatically 

Android 14 and up:
https://developer.android.com/about/versions/14/features#non-linear-font-scaling

Ionic Dynamic Font Scaling:
https://ionicframework.com/docs/layout/dynamic-font-scaling 

## Describe the solution you'd like

https://www.figma.com/design/KRnTge4tZYtAqY2LD9uuC6/Text-resize?node-id=2125-3142

## Are there any additional context?

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#the-process-of-contributing) correctly.

### Refinement

- [ ] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#ux-refinement)
- [ ] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](../CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".
